plugin/pkg/scheduler/factory/factory.go

```diff
@@ -478,48 +478,48 @@ func getNodeConditionPredicate() cache.NodeConditionPredicate {
 // scheduled.
 func (factory *ConfigFactory) createUnassignedNonTerminatedPodLW() *cache.ListWatch {
 	selector := fields.ParseSelectorOrDie("spec.nodeName==" + "" + ",status.phase!=" + string(api.PodSucceeded) + ",status.phase!=" + string(api.PodFailed))
-	return cache.NewListWatchFromClient(factory.Client, "pods", api.NamespaceAll, selector)
+	return cache.NewListWatchFromClient(factory.Client.Core().GetRESTClient(), "pods", api.NamespaceAll, selector)
```

Comment:
Why not just build ListWatch function with `factory.Client.Core().Pod(api.NamespaceAll).List(selecor)` and `factory.Client.Core().Pod(api.NamespaceAll).Watch(selecor)`? Building ListWatch func atop of RESTClient extraced from a client that's already able to list and watch looks wrong to me.